tests/python/unittest/test_gluon.py

```diff
@@ -2750,6 +2775,17 @@ def test_gluon_param_load():
     net.cast('float16')
     net.load_parameters('test_gluon_param_load.params', cast_dtype=True)
     mx.nd.waitall()
+    
+@with_seed()
+def test_gluon_param_load_dtype_source():
+    net = mx.gluon.nn.Dense(10, in_units=10)
+    net.initialize()
+    net.cast('float16')
```

Comment:
Would you mind adding a case to load INT8 weight?

Reply:
Also suggest for adding test for loading int8 weight. BTW, I just took a quick try and this works well with my internal patch:)

Reply:
@pengzhao-intel @xinyu-intel, thanks for the suggestion, I'm actually unable to find a single layer in Gluon that supports `int8` or `uint8`. 

```python
net = mx.gluon.nn.Conv2D(channels=4, kernel_size=3, in_channels=3)
net.initialize()
net.cast(np.uint8)
net(mx.nd.ones((1,3,224,224), dtype=np.uint8))
```

gives me a 
```text
MXNetError: std::exception
```

Any suggestions without resorting to symbolically fused graphs?

Reply:
I have added a test for the `ParameterDict` that should be sufficient since it is the same mechanism as the network parameters: see https://github.com/apache/incubator-mxnet/pull/15230/files#diff-962bd5bb7248659d7eb3be37ee8a4c6bR141